lib/prefab/config_client.rb

```diff
@@ -171,6 +176,36 @@ def load_configs(configs, source)
       finish_init!(source, project_id)
     end
 
+    def cache_path
+      return @cache_path unless @cache_path.nil?
+
+      @cache_path ||= File.join(ENV.fetch('XDG_CACHE_HOME', File.join(Dir.home, '.cache')), "prefab.cache.#{@base_client.options.api_key_id}.json")
+      FileUtils.mkdir_p(File.dirname(@cache_path))
+      @cache_path
+    end
+
+    def cache_configs(configs)
+      return unless @options.use_local_cache && !@options.is_fork
+      File.open(cache_path, "w") do |f|
+        f.flock(File::LOCK_EX)
```

Comment:
This prevents simultaneous writes but doesn't fix the partial read problem
do at least one of
1) f.flock(File:LOCK_SH) on read so the reads are blocked on write completing https://apidock.com/ruby/v2_5_5/File/flock
2) the atomic write (write and mv) (this will still need the write lock)

Reply:
👍 addressed in loader with LOCK_SH